### ISSUE TITLE
Inconsistent indentation with different array notations in ScopeIndent and exact indenting

### ISSUE THREAD

**Issue by Riimu** [squizlabs/PHP_CodeSniffer#1128]
I'm not sure which handling is the exact bug, but seems there is some inconsistency in indentation handling between `array()` and `[]` syntax in `2.6.2`. When using the `Generic.WhiteSpace.ScopeIndent` and setting `exact` to true, the indent is differently identified for these two syntaxes.

For example, having a `ruleset.xml` like:

``` xml
<?xml version="1.0"?>
<ruleset name="Inconsistent">
 <rule ref="Generic.WhiteSpace.ScopeIndent">
  <properties>
   <property name="indent" value="4"/>
   <property name="exact" value="true"/>
  </properties>
 </rule>
</ruleset>
```

And a PHP file like:

``` php
<?php

$array = array(
    'foo' => 'bar',
);

$array = [
    'foo' => 'bar',
]
```

Will result in the following output from phpcs:

```
----------------------------------------------------------------------
FOUND 1 ERROR AFFECTING 1 LINE
----------------------------------------------------------------------
 8 | ERROR | [x] Line indented incorrectly; expected 0 spaces, found
   |       |     4
----------------------------------------------------------------------
PHPCBF CAN FIX THE 1 MARKED SNIFF VIOLATIONS AUTOMATICALLY
----------------------------------------------------------------------
```

Either both array notations should provide the error or neither should provide the error. This way of handling indentation scopes is rather inconsistent.

**Comment by gsherwood** [comment]
As a general note: you typically don't want to use that exact option. Code doesn't work like that in the real world and it is likely to annoy you. It's an old option that someone wanted at a time when I should have said "no" but didn't.

This particular sniff has code to ignore array indentation because arrays are indented in a number of different ways. In this particular case, it sees that line 8 is indented, but has no idea what the indent should be because it is inside an array. It knows, based on the closing brace of the previous array, that the indent should be at least 0 spaces. But the exact option kicks in and expects *exactly* 0 spaces.

I'm not really sure how best to attempt fixing this, or if it is even worth it. I strongly suggest *not* using that exact option because this sniff can't possible test and decent sized code base and not report false positives like this.

**Comment by mj4444ru** [comment]
When using `exact` for `PSR2` standard, we obtain the following:

Source file:
```PHP
<?php
function test()
{
    $data = [
        'test1' => [
            'test11' =>
            [
                'test111' => 1,
            ],
            'test12' =>
            [
                'test121' => 1,
            ],
        ],
    ];
}
```
Will result in the following output from phpcbf:
```PHP
function test()
{
    $data = [
    'test1' => [
    'test11' =>
    [
    'test111' => 1,
            ],
'test12' =>
[
'test121' => 1,
            ],
        ],
    ];
}
```
Source file:
```PHP
<?php
function test()
{
    $data = [
        'test1' => [
            'test11' => [
                'test111' => 1,
            ],
            'test12' => [
                'test121' => 1,
            ],
        ],
    ];
}
```
Will result in the following output from phpcbf:
```PHP
<?php
function test()
{
    $data = [
    'test1' => [
    'test11' => [
    'test111' => 1,
            ],
    'test12' => [
    'test121' => 1,
            ],
        ],
    ];
}
```
Source file:
```PHP
<?php
function test()
{
    $data = array(
    'test1' => array(
    'test11' => array(
    'test111' => 1,
    ),
    'test12' => array(
    'test121' => 1,
    ),
    ),
    );
}
```
Will result in the following output from phpcbf: `No fixable errors were found`

I would like to see a more equal and correct behavior.
